macOS missing memset_s()

### Build System Used
I build wxWidgets and/or my application using:

- [ x ] CMake
- [ ] configure
- [ ] mingw32-make with makefile.gcc
- [ ] MSBuild (Microsoft Visual Studio solution file)
- [ ] nmake with makefile.vc
- [ ] Xcode

### Description
When building wxWidgets for macOS as a submodule, the `memset_s()` function is not found in utilsunx.cpp. This was reported in #19334 but never fully tracked down.
```
~/gui/wxwidgets/src/unix/utilsunx.cpp:231:5: error: use of undeclared identifier 'memset_s'
    memset_s(v, n, 0, n);
    ^
```

https://github.com/wxWidgets/wxWidgets/commit/149db36bc85b2fb2c05e51dbbbb5acfe9b9b3b00 moved `#define __STDC_WANT_LIB_EXT1__ 1` to the top of utilsunx.cpp so that it would be defined before `string.h`; however, (at least on macOS) the PCH header has already transitively includes `string.h`. Since `wxBUILD_PRECOMP` defaults to `On`, the PCH is generated with `string.h` but *without* `memset_s`, which causes the issue.

I believe the fix is that we must define __STDC_WANT_LIB_EXT1__ early in the PCH header so that if anything transitively includes `string.h` we correctly define the `memset_s` symbol, e.g.

```diff
diff --git a/include/wx/wxprec.h b/include/wx/wxprec.h
index 366f0a36fc..a4b4859674 100644
--- a/include/wx/wxprec.h
+++ b/include/wx/wxprec.h
@@ -23,6 +23,10 @@
 
 #ifdef WX_PRECOMP
 
+// Define this as soon as possible and before string.h is included to get
+// memset_s() declaration from it if available.
+#define __STDC_WANT_LIB_EXT1__ 1
+
 // include "wx/chartype.h" first to ensure that UNICODE macro is correctly set
 // _before_ including <windows.h>
 #include "wx/chartype.h"

``` 

CMakeLists.txt looks like this:

```
cmake_minimum_required(VERSION 3.17)

set(CMAKE_EXPORT_COMPILE_COMMANDS On) 
set(CMAKE_CXX_STANDARD 14) 
set(CMAKE_CXX_EXTENSIONS Off)

project(test)

set(wxBUILD_SHARED OFF)
set(wxBUILD_INSTALL OFF)
# Uncomment the below line and things work fine
# set(wxBUILD_PRECOMP OFF)

add_subdirectory(wxwidgets)

add_executable(app app.cpp)

target_link_libraries(app PRIVATE wx::net wx::core wx::base)
```
<!-- Describe the problem here -->


### Platform and version information

- wxWidgets version <!-- [e.g. 3.2.1] --> you are building: origin/master
- wxWidgets port <!-- [e.g. wxMSW, wxGTK, wxOSX] --> you are building: wxMAC
- OS <!-- [e.g. Windows 10, Ubuntu 22.10, macOS 15] --> and its version: macOS 14.5
- Compiler <!-- [e.g. MSVS 2022, gcc 12.1] --> being used: Apple clang version 15.0.0
- Non-default compiler options, if any <!-- e.g. -std=c++26 -->: See above CMakeLists.txt

--- a/include/wx/platform.h
+++ b/include/wx/platform.h
@@ -269,6 +269,12 @@
 #        if !defined(wxSIZE_T_IS_UINT) && !defined(wxSIZE_T_IS_ULONG)
 #            define wxSIZE_T_IS_ULONG
 #        endif
+
+        /* Define this as soon as possible and before string.h is included to
+           get memset_s() declaration from it if available. */
+#       ifndef __STDC_WANT_LIB_EXT1__
+#           define __STDC_WANT_LIB_EXT1__ 1
+#       endif
 #    endif
 
 /*

--- a/src/unix/utilsunx.cpp
+++ b/src/unix/utilsunx.cpp
@@ -18,10 +18,6 @@
 // for compilers that support precompilation, includes "wx.h".
 #include "wx/wxprec.h"
 
-// Define this as soon as possible and before string.h is included to get
-// memset_s() declaration from it if available.
-#define __STDC_WANT_LIB_EXT1__ 1
-
 #include "wx/utils.h"
 
 #if !defined(HAVE_SETENV) && defined(HAVE_PUTENV)

